src/client/components/responsive-video/responsive-video.vue

```diff
@@ -1,17 +1,97 @@
 <template>
-  <figure>
-    <a :href="video.url" target="_blank" rel="noopener" @click="play()">
-      <img alt="" :src="video.thumbnailUrl" :width="video.width" :height="video.height" />
-    </a>
-    <figcaption v-if="video.title">{{ video.title }}</figcaption>
-  </figure>
+  <div class="responsive-video">
+    <figure>
+      <fixed-ratio :content="video">
+        <lazy-load :url="video.thumbnailUrl">
+          <img
+            slot-scope="imageSource"
+            class="responsive-video__image"
+            v-if="!playing && imageSource.computed"
+            alt=""
+            :src="imageSource.computed"
+          />
+        </lazy-load>
+        <iframe
+          v-if="playing"
+          :src="videoSource"
+          style="width:100%; height:100%;"
+          frameborder="0"
+          webkitallowfullscreen
+          mozallowfullscreen
+          allowfullscreen
+          allow="autoplay">
+        </iframe>
+      </fixed-ratio>
+      <figcaption v-if="video.title">
+        <a :href="video.url" target="_blank" rel="noopener">
+          {{ video.title }}
+        </a>
+      </figcaption>
+    </figure>
+    <button v-if="!playing" class="responsive-video__button" @click="play()">
```

Comment:
I'd like this button to:
* be the full size of the video/cover so that you can't miss it when clicking.
* have a focus & hover style so that it's extra clear you can push play.
* be an anchor linking to `video.url` by default, so that it works by default and is progressively enhanced to play in the iframe.